src/main/java/org/kcctl/service/KafkaConnectException.java

```diff
@@ -28,11 +28,16 @@ public static KafkaConnectException from(Response response) {
 
         ErrorResponse errorResponse = response.readEntity(ErrorResponse.class);
 
-        return switch (response.getStatusInfo().toEnum()) {
-            case CONFLICT -> new KafkaConnectConflictException(errorResponse.message);
-            case NOT_FOUND -> new KafkaConnectNotFoundException(errorResponse.message);
-            default -> new KafkaConnectException(errorResponse);
-        };
+        if (response.getStatusInfo() == Response.Status.CONFLICT
+                || errorResponse.message.contains("Request cannot be completed because a rebalance is expected")) {
```

Comment:
Ugh... turns out there's an ancient bug in Connect where the "rebalance pending" response can have either a 409 status code (correct) or a 500 status code (incorrect). Best we can do for now is go back to manually examining the error message to handle that case.

We could technically change this behavior in Connect, and it is probably worth addressing, but for tools like this one that strive for backwards compatibility with a wide range of versions, that won't buy us much for major/minor versions that are already EOL.

Reply:
Gasp, yeah, reality just sucks sometimes ;) But actually also nice demonstration of the value of setting up this matrix test.